src/nodes/Node.ts

```diff
@@ -7,7 +7,20 @@ import type { Scalar } from './Scalar.js'
 import type { YAMLMap } from './YAMLMap.js'
 import type { YAMLSeq } from './YAMLSeq.js'
 
-export type Node = Alias | Scalar | YAMLMap | YAMLSeq
+export type Node<T = unknown> =
+  | Alias
+  | Scalar<T>
+  | YAMLMap<unknown, T>
+  | YAMLSeq<T>
+
+/** Utility type mapper */
+export type NodeType<TValue> = TValue extends string | number | bigint | boolean | null
+  ? Scalar<TValue>
+  : TValue extends Array<any>
+  ? YAMLSeq<NodeType<TValue[number]>>
+  : TValue extends { [key: string | number]: any }
+  ? YAMLMap<NodeType<keyof TValue>, NodeType<TValue[keyof TValue]>>
+  : Node
```

Comment:
Any particular reason to prefer `TValue` rather than something possibly simpler like `Value`?

Reply:
`Value` is already used in other places of the codebase, so I thought it would make sense to distinguish the generic parameters — would make relevant find and rename operations easier.